stream_alert/alert_processor/main.py

```diff
@@ -40,68 +41,41 @@ def handler(event, context):
     """
-    records = event.get('Records', [])
-    LOGGER.info('Running alert processor for %d records', len(records))
-
-    # A failure to load the config will log the error in load_output_config and return here
+    # A failure to load the config will log the error in load_output_config
+    # and return here
     config = _load_output_config()
     if not config:
         return
 
     region = context.invoked_function_arn.split(':')[3]
     function_name = context.function_name
 
-    status_values = []
-
-    for record in records:
-        sns_payload = record.get('Sns')
-        if not sns_payload:
-            continue
-
-        sns_message = sns_payload['Message']
-        try:
-            loaded_sns_message = json.loads(sns_message)
-        except ValueError as err:
-            LOGGER.error('An error occurred while decoding message to JSON: %s', err)
-            continue
-
-        if not 'default' in loaded_sns_message:
-            # do not log for messages related to alarms
-            if not 'AlarmName' in loaded_sns_message:
-                LOGGER.error('Malformed SNS: %s', loaded_sns_message)
-            continue
-
-        status_values.extend(run(loaded_sns_message, region, function_name, config))
+    # Yield back the current status to the caller
+    for status in run(event, region, function_name, config):
+        yield status
 
-    # Return the current status back to the caller
-    return status_values
 
-def run(loaded_sns_message, region, function_name, config):
+def run(alert, region, function_name, config):
     """Send an Alert to its described outputs.
 
     Args:
-        loaded_sns_message [dict]: SNS message dictionary with the following structure:
-
-        {
-            'default': alert
-        }
-
-        The alert is another dict with the following structure:
-
-        {
-            'record': record,
-            'metadata': {
-                'rule_name': rule.rule_name,
-                'rule_description': rule.rule_function.__doc__,
-                'log': str(payload.log_source),
-                'outputs': rule.outputs,
-                'type': payload.type,
-                'source': {
-                    'service': payload.service,
-                    'entity': payload.entity
+        alert [dict]: dictionary representating an alert with the
+            following structure:
+
+            {
+                'record': record,
+                'metadata': {
+                    'rule_name': rule.rule_name,
+                    'rule_description': rule.rule_function.__doc__,
+                    'log': str(payload.log_source),
+                    'outputs': rule.outputs,
+                    'type': payload.type,
+                    'source': {
+                        'service': payload.service,
+                        'entity': payload.entity
+                    }
                 }
             }
-        }
 
         region [string]: the AWS region being used
```

Comment:
let's update this to: `The AWS region of the currently executing Lambda function`